src/Microsoft.Azure.WebJobs.Host/Executors/FunctionInvoker.cs

```diff
@@ -48,6 +48,9 @@ public IReadOnlyList<string> ParameterNames
 
         public async Task InvokeAsync(object[] arguments)
         {
+            // Return a task immediately in case the method is not async.
+            await Task.Yield();
```

Comment:
I see. Yeah, since the config needs to be created before the host, other implementors can't really wire it up in a clean way. Ok for now - might improve later.